src/components/MAPI/clusters/ClusterDetail/utils.ts

```diff
@@ -105,3 +106,79 @@ export async function updateClusterLabels(
 
   return capiv1alpha3.updateCluster(httpClient, auth, cluster);
 }
+
+export function getClusterRegionLabel(cluster?: capiv1alpha3.ICluster) {
+  if (!cluster) return undefined;
+
+  switch (cluster.spec?.infrastructureRef?.kind) {
+    case capzv1alpha3.AzureCluster:
+      return 'Azure region';
+
+    // TODO(axbarsan): Use CAPA type once available.
+    case 'AWSCluster':
+      return 'AWS region';
+
+    default:
+      return '';
+  }
+}
+
+export function getClusterAccountIDLabel(cluster?: capiv1alpha3.ICluster) {
+  if (!cluster) return undefined;
+
+  switch (cluster.spec?.infrastructureRef?.kind) {
+    case capzv1alpha3.AzureCluster:
+      return 'Subscription ID';
+
+    // TODO(axbarsan): Use CAPA type once available.
+    case 'AWSCluster':
+      return 'Account ID';
+
+    default:
+      return '';
+  }
+}
+
+export function getClusterAccountIDPath(
+  cluster?: capiv1alpha3.ICluster,
+  accountID?: string
+) {
+  if (!cluster || !accountID) return undefined;
+
+  switch (cluster.spec?.infrastructureRef?.kind) {
+    case capzv1alpha3.AzureCluster:
+      return 'https://portal.azure.com/';
+
+    // TODO(axbarsan): Use CAPA type once available.
+    case 'AWSCluster':
+      return `https://${accountID}.signin.aws.amazon.com/console`;
+
+    default:
+      return '';
+  }
+}
+
+export function getCredentialsAccountID(
+  credentials?: legacyCredentials.ICredential[]
+) {
+  if (!credentials) return undefined;
+  if (credentials.length < 1) return '';
+
+  const mainCredential = credentials.find((credential, _, collection) => {
```

Comment:
We use the first custom credential in the list if it exists, otherwise we use the default credential